code/game/machinery/doors/firedoor.dm

```diff
@@ -61,17 +62,16 @@
 	. = ..()
 	COOLDOWN_START(src, detect_cooldown, DETECT_COOLDOWN_STEP_TIME)
 	soundloop = new(src, FALSE)
-	AddElement(/datum/element/atmos_sensitive, mapload)
 	CalculateAffectingAreas()
-	my_area = get_area(src)
+	AddElement(/datum/element/atmos_sensitive, mapload)
 	var/static/list/loc_connections = list(
 		COMSIG_TURF_EXPOSE = .proc/check_atmos,
 	)
-
 	AddElement(/datum/element/connect_loc, loc_connections)
```

Comment:
Do you really need the connect loc element either? it's basically what you're doing manually with CalculateWatchedTurfs and Moved